mesonbuild/build.py

```diff
@@ -1934,8 +1934,13 @@ def post_init(self) -> None:
                 mlog.debug('Defaulting Rust static library target crate type to rlib')
                 self.rust_crate_type = 'rlib'
             # Don't let configuration proceed with a non-static crate type
-            elif self.rust_crate_type not in ['rlib', 'staticlib']:
-                raise InvalidArguments(f'Crate type "{self.rust_crate_type}" invalid for static libraries; must be "rlib" or "staticlib"')
+            elif self.rust_crate_type not in ['rlib', 'staticlib', 'object']:
+                raise InvalidArguments(f'Crate type "{self.rust_crate_type}" invalid for static libraries; must be "rlib", "staticlib" or "object"')
+            elif self.rust_crate_type == 'object':
+                mlog.warning("""Due to the unreliability of Rustc, crate type "object" compiled output might be unstable and require extra work to use.
+The use of this feature means that Meson makes no guarantee anything works. We believe it's likely this is going to break without warning in
+micro-releases of rust, and if it does, you need to report it to the package making use of it, not Meson, because all bugs are the responsibility of
+the package using unstable features.""")
```

Comment:
This kind of dedenting makes hash of vim's ability to fold code, can we do:
```suggestion
                mlog.warning(textwrap.dedent("""\
                    Due to the unreliability of Rustc, crate type "object" compiled output might be unstable and require extra work to use.
                   The use of this feature means that Meson makes no guarantee anything works. We believe it's likely this is going to break without warning in
                   micro-releases of rust, and if it does, you need to report it to the package making use of it, not Meson, because all bugs are the responsibility of
                   the package using unstable features."""))
```

Reply:
Sure, updated